pkg/api/v1/clients/multicluster/multi_cluster_resource_client_kube_test.go

```diff
@@ -0,0 +1,239 @@
+package multicluster_test
+
+import (
+	"context"
+	"log"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/solo-io/go-utils/kubeutils"
+	kubehelpers "github.com/solo-io/go-utils/testutils/kube"
+	skapi "github.com/solo-io/solo-kit/api/multicluster/v1"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/factory"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/kube"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/cache"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/clientgetter"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/multicluster"
+	"github.com/solo-io/solo-kit/pkg/api/v1/clients/wrapper"
+	"github.com/solo-io/solo-kit/pkg/api/v1/resources/core"
+	multicluster2 "github.com/solo-io/solo-kit/pkg/multicluster"
+	"github.com/solo-io/solo-kit/pkg/multicluster/clustercache"
+	"github.com/solo-io/solo-kit/pkg/multicluster/secretconverter"
+	skpkg "github.com/solo-io/solo-kit/pkg/multicluster/v1"
+	"github.com/solo-io/solo-kit/test/helpers"
+	v1 "github.com/solo-io/solo-kit/test/mocks/v1"
+	"github.com/solo-io/solo-kit/test/testutils"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+)
+
+var _ = Describe("MultiClusterResourceClient e2e test", func() {
+	if os.Getenv("RUN_KUBE_TESTS") != "1" {
+		log.Printf("This test creates kubernetes resources and is disabled by default. To enable, set RUN_KUBE_TESTS=1 in your env.")
+		return
+	}
+
+	var (
+		namespace       string
+		subject         v1.AnotherMockResourceClient
+		watchAggregator wrapper.WatchAggregator
+		localRestConfig *rest.Config
+		localKubeClient kubernetes.Interface
+		ctx             context.Context
+		cancel          context.CancelFunc
+		cfgSecret       *corev1.Secret
+		remoteCluster   = "cluster-two"
+		// resources on different clusters can have the same name and namespace
+		resourceName = "name"
+	)
+
+	getAnotherMockResource := func(cluster, ns, name, basicField string) *v1.AnotherMockResource {
+		return &v1.AnotherMockResource{
+			Metadata: core.Metadata{
+				Name:      name,
+				Namespace: ns,
+				Cluster:   cluster,
+			},
+			BasicField: basicField,
+		}
+	}
+
+	BeforeEach(func() {
+		ctx, cancel = context.WithCancel(context.Background())
+		cacheManager, err := clustercache.NewCacheManager(ctx, kube.NewKubeSharedCacheForConfig)
+		Expect(err).NotTo(HaveOccurred())
+		clientGetter := clientgetter.NewKubeResourceClientGetter(
+			cacheManager,
+			v1.AnotherMockResourceCrd,
+			false,
+			nil,
+			0,
+			factory.NewResourceClientParams{ResourceType: &v1.AnotherMockResource{}},
+		)
+		watchAggregator = wrapper.NewWatchAggregator()
+		watchHandler := multicluster.NewAggregatedWatchClusterClientHandler(watchAggregator)
+		clientSet := multicluster.NewClusterClientManager(context.Background(), clientGetter, watchHandler)
```

Comment:
at this point i would consider renaming `ClusterClientManager` to `ClusterClientGetter` and `ClientGetter` to `ClientFactory`

both `manager` and `getter` are not accurate for what these interfaces do

Reply:
made the change from `getter` to `factory` but i prefer the name `manager` for the manager because it both enables getting clients for a cluster and any callbacks on add/remove client. what do you think?

Reply:
`Manager` is vague so i'd prefer something else instead. since the interface only implements `ClientForCluster(cluster string) (client clients.ResourceClient, found bool)`, i think `ClusterClientGetter` makes sense for the interface type

manager is less objectionable as far as the implemented type, but i would prefer something that says "this thing GetsClients and Handles Clusters` (ClientGettingClusterHandler?)